numpy/core/tests/test_multiarray.py

```diff
@@ -9989,3 +9995,35 @@ def test_sort_uint():
 
 def test_private_get_ndarray_c_version():
     assert isinstance(_get_ndarray_c_version(), int)
+
+
+@pytest.mark.parametrize("N", np.arange(1, 512))
+@pytest.mark.parametrize("dtype", ['e', 'f', 'd'])
+def test_argsort_float(N, dtype):
+    np.random.seed(42)
+    # (1) Regular data with a few nan: doesn't use vectorized sort
+    arr = -0.5 + np.random.sample(N).astype(dtype)
+    arr[np.random.choice(arr.shape[0], 3)] = np.nan
+    assert_arg_sorted(arr, np.argsort(arr, kind='quick'))
+
+    # random data with inf at the end of array
+    # See: https://github.com/intel/x86-simd-sort/pull/39
+    arr = -0.5 + np.random.sample(N).astype(dtype)
+    arr[N-1] = np.inf
+    assert_arg_sorted(arr, np.argsort(arr, kind='quick'))
+
+
+@pytest.mark.parametrize("N", np.arange(1, 512))
+@pytest.mark.parametrize("dtype", ['h', 'H', 'i', 'I', 'l', 'L'])
+def test_argsort_int(N, dtype):
+    # Random data with MAX and MIN sprinkled
+    minv = np.iinfo(dtype).min
+    maxv = np.iinfo(dtype).max
+    arr = np.random.randint(low=minv, high=maxv-1, size=N, dtype=dtype)
```

Comment:
Maxv is not in the array generated.
```suggestion
    arr = np.random.randint(low=minv, high=maxv, size=N, dtype=dtype)
```
But that is not entirely correct either, since randint uses a half open interval.
Maybe maxv+1 will work, but is depends on the type.

Maybe just update the docstring, not sure whether it is important maxv is really in the array, or just some other high value.

Reply:
`maxv+1` will work. We don't convert from the unbounded Python `int` object to the `dtype` until after we check for just this case.

But that doesn't really matter as it is vanishingly unlikely that either `minv` or `maxv` will be in the array for the larger integer dtypes. Instead, you'll need to mix them in explicitly as one did with `inf` in `test_argsort_float`.

```suggestion
    rnd = np.random.RandomState(1100710816)
    arr = rnd.randint(low=minv, high=maxv+1, size=N, dtype=dtype)
    i, j = rnd.choice(N, 2, replace=False)
    arr[i] = minv
    arr[j] = maxv
```

Reply:
Makes sense.